source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -398,6 +398,12 @@ Http::FilterTrailersStatus JsonTranscoderFilter::encodeTrailers(Http::HeaderMap&
     response_headers_->insertGrpcMessage().value(*grpc_message_header);
   }
 
+  // remove Trailer headers if the client connection was http/1
+  if (encoder_callbacks_->requestInfo().protocol() != Http::Protocol::Http2) {
+    static const Http::LowerCaseString trailerKey = Http::LowerCaseString("trailer");
```

Comment:
nit: `trailer_key` per style guide.

Reply:
ah, sorry. been to deep into Go lately...